### ISSUE TITLE
alternative pins for ESP32

### ISSUE THREAD

**Issue by rodrigtti** [felis/USB_Host_Shield_2.0#656]
Hi guys , in my project i need to use other pins to SPI interface for ESP32 I tried to edit the avrpins.h as the example bellow with no luck

`MAKE_PIN(P19, 12); // MISO
MAKE_PIN(P23, 13); // MOSI
MAKE_PIN(P18, 14); // SCK`

reading the esp32 docs (https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/peripherals/spi_master.html) I found out that it has 2 general purpose SPI interface that we could use ,one is HSPI with pins are( 18,19,23) as same as it is proposed on this lib, and  VSPI (pins 14 ,12,13).
So my question is is possible to change HSPI to VSPI?

I also tried to change some config on SPI.H with no success

I changed this line :
`  SPIClass(uint8_t spi_bus=HSPI);`

to:

`  SPIClass(uint8_t spi_bus=VSPI);`

resulting the same message "OSC did not start"

**Comment by DrJaymz** [comment]
I'm trying to do this as I don't have those pins available.  Both HSPI and VSPI can be mapped to other pins however for some reason in the overly complicated its done in this library its not that simple.

**Comment by joshkanna** [comment]
did you ever end up figuring out how to do this?

**Comment by DrJaymz** [comment]
> did you ever end up figuring out how to do this?

Yes.

#define PIN_SPI_SCK 33 // SCK
#define PIN_SPI_MISO 36 // SCL
#define PIN_SPI_MOSI 5 // SDA
#define PIN_SPI_SS 32 // SS

I got it working on these pins.  I remember it being a real pig and I couldn't for the life of me work out why you couldn't just call a constructor with the pins.  I needed to get it working at the same time as ethernet and SPI TFT display.  In the end that works but the order of initialisation seemed to matter.

Looking at it, I have changed usbhost.h  My changes on the right:

<img width="1347" height="518" alt="Image" src="https://github.com/user-attachments/assets/a4ff7a5f-68d8-492b-bc5f-7170d75f37ab" />

<img width="1408" height="215" alt="Image" src="https://github.com/user-attachments/assets/994a4468-ea70-45ea-8c07-832065989c1f" />

So it looks like I tried to use the defines and then ended up having to chase it around with these pin defs.

Also see: https://github.com/felis/USB_Host_Shield_2.0/issues/822#issuecomment-2249533326

TLDR: the library "helpers" didn't help and all you really needed to do was:
in usbhost.h: USB_SPI.begin(33,36,5,32); <-- changed to the pins I wanted.
Instead of calling something else then something else then something else then something else in files where everything is conditionally compiled 22 levels deep; its like that to get it to work on every controller under the sun, but some of us have work to do...

Forget setting V or H SPI (which I also tried), I just init my TFT then USB then ETH and whatever it does they all work together on 3 separate SPI busses.
